server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -152,25 +153,21 @@ public Response getServer(@PathParam("id") String queryId, @Context final HttpSe
       log.debug("Received cancel request for query [%s]", queryId);
     }
     if (authConfig.isEnabled()) {
-      // This is an experimental feature, see - https://github.com/druid-io/druid/pull/2424
-      final AuthorizationInfo authorizationInfo = (AuthorizationInfo) req.getAttribute(AuthConfig.DRUID_AUTH_TOKEN);
-      Preconditions.checkNotNull(
-          authorizationInfo,
-          "Security is enabled but no authorization info found in the request"
-      );
       Set<String> datasources = queryManager.getQueryDatasources(queryId);
       if (datasources == null) {
         log.warn("QueryId [%s] not registered with QueryManager, cannot cancel", queryId);
-      } else {
-        for (String dataSource : datasources) {
-          Access authResult = authorizationInfo.isAuthorized(
-              new Resource(dataSource, ResourceType.DATASOURCE),
-              Action.WRITE
-          );
-          if (!authResult.isAllowed()) {
-            return Response.status(Response.Status.FORBIDDEN).header("Access-Check-Result", authResult).build();
-          }
-        }
+        datasources = Sets.newTreeSet();
```

Comment:
I think it's easier to keep it as-is, sinceAuthorizationUtils.authorizeAllResourceActions() needs to be called to set the request headers properly (otherwise the PreResponseAuthorizationCheckFilter will later think that the request was never authorized).